lib/utils/tunnel.ts

```diff
@@ -15,10 +15,13 @@ limitations under the License.
 */
 import type { BalenaSDK } from 'balena-sdk';
 import { Socket } from 'net';
+import * as tls from 'tls';
 import { TypedError } from 'typed-error';
 
 const PROXY_CONNECT_TIMEOUT_MS = 10000;
 
+class TunnelServerNotTrustedError extends TypedError {}
```

Comment:
As coded, `new TunnelServerNotTrustedError(...)` will cause the CLI to report this error to Sentry.io. I assume that such errors do not represent CLI issues that need fixing, rather represent issues in the user's local network -- which would be reasonable for users to bring to support attention, at which point support would investigate the usual Deep Packet Inspection, SSL interception issues in the local network. If you agree with this reasoning, then the way to avoid the error being reported to Sentry.io would be to extend `ExpectedError`:
```
import { ExpectedError } from '../errors';

class TunnelServerNotTrustedError extends ExpectedError {}
```

Reply:
What happens to the UX, the same?

Reply:
Here's how the global error handler treats the error depending on whether or not it is "expected":
https://github.com/balena-io/balena-cli/blob/v12.38.3/lib/errors.ts#L247-L262

Reply:
OK, so UX is the same.

I leave it to you @pdcastro to make that call then; personally I would rather see if people are hitting issues, and at what frequency vs good connections, but this is your ship dude 😁

Reply:
It depends on what `TunnelServerNotTrustedError` means. My view is that an error reported to Sentry.io is something that we need to fix (and after fixed, it stops being reported). Two extremes are:
* If the user types a command-line option that doesn't exist, e.g. `--nono`, this definitely should not be reported to Sentry.io. It's just noise and counts towards Sentry.io's monthly report limit and max API rate limit that then prevents "legitimate" errors from being reported.
* Properly unexpected and unhandled errors (excluding the catch-all global handler) such as "TypeError" of "startsWith does not exist on undefined" should definitely be reported and it should definitely be fixed - typically a logic error in source code, and often one that blocks users / renders the command unusable.

If `TunnelServerNotTrustedError` means a problem in the user's local network (SSL interception), then it's not normally something actionable by balena, and becomes noise in Sentry.io -- noise that in the past resulted in CLI pull requests to replace `Error` or `TypedError` with `ExpectedError`, so that it stops being reported. However, if we had the view that `TunnelServerNotTrustedError` signified something actionable such as, say, the server certificate has expired and we need to update it, then we'd want it reported in Sentry.io.  But dev ops takes care of replacing certificates when they expire, especially at the ELB/NLB level.

`TunnelServerNotTrustedError` feels like a grey area. We will be testing that `balena tunnel` works before the PR is merged, and on balance I don't think it should be reported to Sentry.io. If it's for me to decide, it's an `ExpectedError`. :-)  (I also take into account the fact that there are plenty of bugs and longstanding issues reported by users themselves, let alone Sentry.io, which are known and in the queue to be fixed, so we don't need additional reports of things that most likely aren't even actionable...)